website/common/script/libs/shops.js

```diff
@@ -155,7 +155,16 @@ shops.getMarketGearCategories = function getMarketGear (user, language) {
       text: getClassName(classType, language),
     };
 
-    let result = filter(content.gear.flat, ['klass', classType]);
+    let result = filter(content.gear.flat, function findClassGear (gearItem) {
+      if (gearItem.klass === classType) return true;
+      let classShift = {
+        items: user.items,
+        stats: {
+          class: classType,
+        },
+      };
+      if (gearItem.specialClass === classType) return gearItem.canOwn(classShift);
```

Comment:
this should also merge the `specialGear` https://github.com/HabitRPG/habitica/blob/c1bd762ff159112e5b735b387f3500af94c60280/website/common/script/libs/shops.js#L175
which adds the special gear that has been already owned


once both merged it should also check about the current season gear. Cause I don't think we should list gear that can't be purchased at the moment (which also never been owned)

Reply:
OK, do you suggest just chaining (or following up with) a Lodash `.merge` calling that function then?

This won't list gear that can't be purchased. `canOwn` takes care of that part.